### PR TITLE
HW 7

### PR DESCRIPTION
**Pull request by saharj**

### CODE REVIEW

File: sahar.saharj/hw_7/homework/script.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+(function() {
+
+	($('#zip').on('keyup', function() {
+		var zipCode = $('#zip').val();
+		var url = 'http://api.zippopotam.us/us/' + zipCode;
+
+		console.log('url: ' + url);
+		process(url, cityAndState);
+	}))();
```
Comment (Borgaard) [inline sahar.saharj/hw_7/homework/script.js:24]: This is the one we just worked on during office hours, right? The event handler code still worked after turning it into an IIFE. Yay!

How comfortable do you feel with IIFEs and their purpose now?

File: sahar.saharj/hw_7/homework/script.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			},
+			error: function() {
+				console.log('There was an error occured');
+			}
+		});
+	};
+
+	function cityAndState(r) {
+		var city = r.places[0]['place name'];
+		var state = r.places[0]['state abbreviation'];
```
Comment (Borgaard) [inline sahar.saharj/hw_7/homework/script.js:40]: Interesting that Zippopotam used keys with spaces (such as `'state abbreviation'`) in their API as that can sometimes be frustrating for developers, but you handled it well using the `['target bracket']` syntax.

At some point you'll make your own API and can give developers the option of using dot syntax or target bracket syntax. 😋

File: sahar.saharj/hw_7/homework/script.js (hunk 1 above)
Comment (saharj) [reply sahar.saharj/hw_7/homework/script.js:24]: At first my understanding about IIFEs was around 30%, and now it is maybe 65%. It is not very high, but I'm happy about the progress.

File: sahar.saharj/hw_7/homework/script.js (hunk 2 above)
Comment (saharj) [reply sahar.saharj/hw_7/homework/script.js:40]: Thank you! :)

**Pull request merged**